File: 서인/CH06/7_교육과정_설계.js
function solution(need, plan){
    let answer = "YES";
    let queue = need.split('');

    for (let x of plan){
      if (queue.includes(x)) {
        if (queue[0] !== x) return "NO"
        else queue.shift(x)
      }
    }

    if(queue.length > 0) return "NO";  
    return answer;
}

let a="CBA";
let b="CBDAGE";
console.log(solution(a, b));

File: 종복/Java/Stack&Queue/7. 교육 과정 설계.java
public String solution(String need, String plan){
		String answer="YES";
	    	
	    ArrayList<Integer> tmp = new ArrayList<>(); 	
        
        for(char x: need.toCharArray()){
            int pos = plan.indexOf(x);
            tmp.add(pos);
        }
        
        System.out.println(tmp);
        for(int i=0; i<tmp.size()-1; i++){
            if(tmp.get(i) > tmp.get(i+1)){
                return "NO";
            }
        }
        
      
		return answer;
	}
